lib/open_food_network/user_balance_calculator.rb

```diff
@@ -6,13 +6,29 @@ def initialize(email, distributor)
     end
 
     def balance
-      -completed_orders.sum(&:outstanding_balance)
+      payment_total - completed_order_total
     end
 
     private
 
+    def completed_order_total
+      completed_orders.sum &:total
+    end
+
+    def payment_total
+      payments.sum &:amount
+    end
+
     def completed_orders
-      Spree::Order.where(distributor_id: @distributor, email: @email).complete.not_state(:canceled)
+      all_orders.complete.not_state(:canceled)
+    end
+
+    def all_orders
+      Spree::Order.where(distributor_id: @distributor, email: @email)
+    end
+
+    def payments
```

Comment:
I'd add a comment to def payments
 `# Lists all complete user payments including payments in incomplete or canceled orders`